dot/core/service.go

```diff
@@ -551,3 +566,30 @@ func (s *Service) GetReadProofAt(block common.Hash, keys [][]byte) (
 
 	return block, proofForKeys, nil
 }
+
+// getRuntimeFromDB gets the runtime for the corresponding block hash from storageState
+func (s *Service) getRuntimeFromDB(blockHash *common.Hash) (instance runtime.Instance, err error) {
```

Comment:
Since this only uses `s.storageState`, should we define it as a method on `StorageState` instead?

Maybe it's not possible due to cycle dependency that I don't know yet.